ocw/lib/gce.py

```diff
@@ -207,3 +215,129 @@ def cleanup_all(self):
+        for email in vault_account_tuple:
+            service_resp = service.projects().serviceAccounts().keys().list(
+                name='projects/-/serviceAccounts/%s' % email).execute()
+            filtered_service_accounts += [k['name'] for k in service_resp['keys'] if dt_frm(
+                    k['validAfterTime'][:-1]) < time_limit]
+        return list(filter(lambda e: e in str(filtered_service_accounts), vault_account_tuple))
+
+    @filterService(name='vaultopenqa')
+    def parser_service_account_list(self, iam_service):
+        ''' Returns a list of the emails of the Service Accounts
+
+        The request is using the resource name of the project associated with
+        the service accounts. This returns a list of dict items, which are
+        also not vault related.
+        The objects returned in chunks which contain a `nextPageToken`
+        to the next page. The `parser_service_account_list` parses each page
+        and returns the full list. Subsequently, the list is filtered to return
+        only emails correlated with vault Service Accounts
+
+        Parameters
+        ----------
+        iam_service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        '''
+        service_accounts_emails = []
+        req = iam_service.projects().serviceAccounts().list(
+            name='projects/suse-sle-qa')
+
+        while True:
+            resp = req.execute()
+            service_accounts_emails += [service_account['email']
+                                        for service_account in resp.get('accounts', [])]
+
+            req = iam_service.projects().serviceAccounts().list_next(
+                previous_request=req, previous_response=resp)
+            if req is None:
+                break
+        return tuple(set(service_accounts_emails))
+
+    def delete_vaultopenqa_service_accounts(self, iam_service, vault_accounts):
+        ''' Deletes a list of vault keys
+
+        Expects _iam_service_ of type *googleapiclient.discovery.Resource*
+        and _vault_accounts_ list of strings.
+
+        A short comment from the documentation. - Deleting a service account
+        key does not revoke short-lived credentials
+        that have been issued based on the service account key.
+
+        Parameters
+        ----------
+        iam_service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        vault_accounts : list, required
+            A list of keys to delete. This is retrieved by the _accounts_ of
+            the *parser_service_account_list* which is the email field.
+
+        Raises
+        ------
+        TypeError
+            This will raised when the _vault_accounts_ will not match the expected pattern in the _name_
+        HttpError
+            Http Respond Errors
+        '''
+        if (len(vault_accounts) < 1):
+            self.log_info("Nothing to delete")
```

Comment:
```suggestion
            self.log_debug("Nothing to delete")
```

Reply:
IMO the whole log_* has wrong application here. i was so tempted to changed it but it is far from the scope of this task.
_debug_ has higher level than _info_. meaning that if the logging was used properly and the logging level was _info_ the debug messages would not showed up. Something that in this particular case it would be wrong as you want to have the information in the regular logs.